src/main/kotlin/org/rust/ide/inspections/RsRedundantElseInspection.kt

```diff
@@ -27,10 +21,23 @@ class RsRedundantElseInspection : RsLocalInspectionTool() {
 
     override fun buildVisitor(holder: RsProblemsHolder, isOnTheFly: Boolean): RsVisitor =
         object : RsVisitor() {
-            override fun visitElseBranch(expr: RsElseBranch) {
-                if (!expr.isRedundant) return
 
-                val elseExpr = expr.`else`
+            override fun visitElseBranch(expr: RsElseBranch) = visitElseOrLetElseBranch(expr)
+
+            override fun visitLetElseBranch(expr: RsLetElseBranch) = visitElseOrLetElseBranch(expr)
+
+            private fun visitElseOrLetElseBranch(expr: RsElement) {
+                when {
```

Comment:
[minor] this might be a bit simpler
```Kotlin
override fun visitElseBranch(expr: RsElseBranch) {
    if (expr.isRedundant) registerProblem(expr, expr.`else`)
}

override fun visitLetElseBranch(expr: RsLetElseBranch) {
    if (expr.isRedundant) registerProblem(expr, expr.`else`)
}
```